src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java

```diff
@@ -458,7 +458,6 @@ private void processIdent(DetailAST ast) {
         }
         switch (parentType) {
             case TokenTypes.ANNOTATION_MEMBER_VALUE_PAIR:
-            case TokenTypes.ANNOTATION:
```

Comment:
This case does nothing. The default does nothing if `checkFields` is `false`. So the only thing that matters here is when `checkFields` is `true`.

For a violation to show up, we need a frame returned from `getFieldWithoutThis` on the AST.

It seems to me we need an annotation that is also named after a field to confuse the logic. (Still investigating)

Reply:
> we need an annotation that is also named after a field to confuse the logic

This seems is what is needed, so I found a test to kill the mutation.